transcendence-app/src/chat/chatroom/infrastructure/db/postgres/chatroom.postgres.repository.ts

```diff
@@ -103,6 +138,104 @@ export class ChatroomPostgresRepository
       : null;
   }
 
+  async getAuthUserPaginatedChatsAndChatrooms(
+    authUserId: string,
+    queryDto: Required<PaginationWithSearchQueryDto>,
+  ): Promise<GenericChat[] | null> {
+    const { limit, offset, search } = queryDto;
+    const messages = await makeQuery<GenericChatData>(this.pool, {
+      text: `
+          WITH msg as (SELECT ARRAY [least(m.${chatMessageKeys.SENDER_ID}, m.${chatMessageKeys.RECIPIENT_ID}),
+                                  greatest(m.${chatMessageKeys.SENDER_ID}, m.${chatMessageKeys.RECIPIENT_ID})] AS "partakers",
+                              m.${chatMessageKeys.ID},
+                              m.${chatMessageKeys.CREATED_AT}
+                       FROM ${table.CHAT_MESSAGE} m
+                       WHERE (m.${chatMessageKeys.SENDER_ID} = $1
+                           OR m.${chatMessageKeys.RECIPIENT_ID} = $1)),
+               dateProvider AS (SELECT msg."partakers",
+                                       ROW_NUMBER() OVER (
+                                           PARTITION BY msg."partakers"
+                                           ORDER BY msg.${chatMessageKeys.CREATED_AT} DESC
+                                           ) AS "rowNumber",
+                                       msg.${chatMessageKeys.ID}
+                                FROM msg),
+               msgIds AS (SELECT dp.${chatMessageKeys.ID} AS "msgId"
+                          FROM dateProvider dp
+                          WHERE dp."rowNumber" = 1),
+               msgData
+                   AS (SELECT CASE WHEN (cm.${chatMessageKeys.SENDER_ID} = $1) THEN cm.${chatMessageKeys.RECIPIENT_ID} ELSE cm.${chatMessageKeys.SENDER_ID} END AS "userId",
+                              u.${userKeys.USERNAME}                                                                                                            AS "lastMsgSenderUsername",
+                              cm.${chatMessageKeys.CONTENT},
+                              cm.${chatMessageKeys.CREATED_AT}
+                       FROM ${table.CHAT_MESSAGE} cm
+                                INNER JOIN msgIds mi ON cm.${chatMessageKeys.ID} = mi."msgId"
+                                LEFT JOIN ${table.USERS} u ON cm.${chatMessageKeys.SENDER_ID} = u.${userKeys.ID}
+                       ORDER BY cm.${chatMessageKeys.CREATED_AT}),
+               lChatMessages as (SELECT u.${userKeys.AVATAR_ID},
+                                        u.${userKeys.AVATAR_X},
+                                        u.${userKeys.AVATAR_Y},
+                                        'chat/' || u.${userKeys.USERNAME}          AS "url",
```

Comment:
Again:
maybe we could pass this as a parameter in the incoming request body so that webapp can own what the route of its own pages are.

Reply:
fixed